examples/multiwindow.rs

```diff
@@ -187,27 +185,30 @@ impl Window {
             api,
             font_instance_key,
             #[cfg(feature = "gfx-hal")]
-            adapter,
-            #[cfg(feature = "gfx-hal")]
             instance,
         }
     }
 
     fn tick(&mut self) -> bool {
-        #[cfg(not(any(feature = "vulkan", feature = "dx12", feature = "metal")))]
+        #[cfg(not(feature = "gfx-hal"))]
         unsafe {
             self.window.make_current().ok();
         }
         let mut do_exit = false;
         let my_name = &self.name;
         let renderer = &mut self.renderer;
-
-        #[cfg(feature = "gfx-hal")]
-        let adapter = &self.adapter;
         #[cfg(feature = "gfx-hal")]
-        let instance = &self.instance;
+        let api = &self.api;
         #[cfg(feature = "gfx-hal")]
-        let window = &self.window;
+        let document_id = self.document_id;
+
+
+        let device_pixel_ratio = self.window.get_hidpi_factor() as f32;
+        let mut framebuffer_size = {
+            let LogicalSize { width, height } = self.window.get_inner_size().unwrap();
+            DeviceUintSize::new(width as u32, height as u32)
+        };
+        let mut layout_size = framebuffer_size.to_f32() / euclid::TypedScale::new(device_pixel_ratio);
```

Comment:
something is wrong here - we shouldn't be dividing `LogicalSize` by HiDPI factor

Reply:
That's totally correct. I sent a [PR](https://github.com/servo/webrender/pull/3008) to fix this in the upstream WR, since the framebuffer size must be `PhysicalSize` not `LogicalSize` in this example.